try#167 - defining repeatable shifts

Taktar opened the pull request:
Fixes #167

What was changed?
ScheduleData.js (previously ScheduleData.tsx)

Renamed from .tsx to .js to match the project’s JavaScript setup.
Contains a data file defining standard (repeatable) shifts with offsets from midnight.
Schedule.jsx (previously Schedule.tsx)

Renamed from .tsx to .jsx and converted to plain JavaScript.
Created a dedicated scheduling view that displays these standard shifts and a calendar.
Inline styles were moved into a separate Schedule.css file.
NavBarShelterDashboard.js

Updated the “Set Shifts” link to point to /set-shifts instead of /request-for-help.
App.js

Added a new route (/set-shifts) under the protected routes so that the shelter admin can access the new scheduling view.
Reverted the VolunteerDashboard import path to ./components/volunteer/VolunteerDashboard to fix compile errors.
Adjusted JSX formatting to satisfy ESLint rules (e.g., react/jsx-newline, indentation).
Schedule.css

New CSS file in client_app/src/styles/shelter/ to hold all styling for the Schedule component (instead of inline styles).
Why was it changed?
Manual Shift Definition: Previously, shelter admins had to define each shift manually for each day, which was time-consuming for common, repeatable shifts (e.g., morning, afternoon, evening).
Repeatable Shifts: The new “Set Shifts” feature allows admins to quickly define multiple repeated shifts by selecting from pre-defined standard shifts and placing them on a calendar.
Usability & Maintainability: Converting .tsx to .jsx and moving inline styles into a .css file aligns better with the existing project setup, simplifies styling, and prevents compile errors in a non-TypeScript environment.
How was it changed?
ScheduleData.js

Added an array of shift objects, each with a start and end offset in milliseconds after midnight (morning, afternoon, evening).
Schedule.jsx

Imports ScheduleData and renders the shifts as clickable cards.
Uses react-big-calendar to display a weekly calendar.
When an admin clicks a shift card, that shift is marked as “active.” Clicking a day in the calendar calculates the shift’s start/end times based on midnight plus the shift’s offsets.
Schedule.css

Moved inline styles from Schedule.jsx to this new CSS file in client_app/src/styles/shelter/.
Provides a cleaner separation of concerns and easier maintenance.
NavBarShelterDashboard.js

Changed the link from /request-for-help to /set-shifts, ensuring the “Set Shifts” button opens the new Schedule component. ./components/volunteer/VolunteerDashboard.
App.js

Imports the new Schedule.jsx and adds a /set-shifts route under ProtectedRoute, restricting access to authenticated shelter admins.
Fixed ESLint errors by adjusting JSX formatting, and reverted the VolunteerDashboard import path to
<img width="1434" alt="Screenshot 2025-02-23 at 3 50 38 PM" src="https://github.com/user-attachments/assets/ae385d79-4dfb-49f8-8779-14a51d8fbdad" />
<img width="1435" alt="Screenshot 2025-02-23 at 3 51 15 PM" src="https://github.com/user-attachments/assets/8ec51621-abc4-4fb8-8970-f77d3ff65f92" />
<img width="462" alt="Screenshot 2025-02-23 at 4 46 31 PM" src="https://github.com/user-attachments/assets/ef539d6b-d087-44a6-a856-f8abd01f0ba4" />

I can click the shifts; here I clicked the Afternoon shift, so it highlighted. But it is not redirecting me somewhere else.

kate-holdener commented on client_app/src/App.js:
```diff
@@ -1,7 +1,8 @@
 import { PastShifts, UpcomingShifts } from "./components/volunteer/Shifts";
 import Shelters from "./components/volunteer/Shelters";
-import VolunteerDashboard from "./components/volunteer/VolunteerDashboard";
+import VolunteerDashboard from "./components/volunteers/VolunteerDashboard";
```
I am getting a compiler error due to this change. See screenshot below
<img width="1207" alt="Screenshot 2025-02-23 at 8 02 50 AM" src="https://github.com/user-attachments/assets/229b071a-859b-4e5b-87a3-9a68261a7511" />

kate-holdener commented on client_app/src/components/shelter/NavBarShelterDashboard.js:
```diff
@@ -1,3 +1,4 @@
+// src/components/shelter/NavBarShelterDashboard.js
```
Please remove this comment

kate-holdener commented on client_app/src/components/shelter/Schedule.tsx:
I am getting an error when I run the client side app that App.js cannot resolve './components/shelter/Schedule'. I wonder if that's because of the .tsx file extension (instead of .js). Error screenshot is below:
<img width="1138" alt="Screenshot 2025-02-23 at 8 09 55 AM" src="https://github.com/user-attachments/assets/99fa8074-ca13-4402-9b5d-1981deca1079" />

When I change the file extension on Schedule.tsx to Schedulue.js, it gets further and then fails with the following error:
<img width="1604" alt="Screenshot 2025-02-23 at 8 12 41 AM" src="https://github.com/user-attachments/assets/6e35550c-4b30-4016-907b-b5bd02f2868a" />

kate-holdener commented on client_app/src/components/shelter/Schedule.tsx:
```diff
@@ -0,0 +1,104 @@
+        {ScheduleData.Content.map((shift) => (
+          <div
+            key={shift.name}
+            style={{
+              display: "inline-block",
+              padding: "0.5rem 1rem",
+              marginRight: "1rem",
+              backgroundColor:
+                activeShiftType === shift.name ? "#d0eaff" : "#f0f0f0",
+              cursor: "pointer",
+              border: "1px solid #ccc",
+              borderRadius: "4px",
```
Please move the styling into a .css file located in the client_app/src/styles directory (or use a style from an existing .css file)

kate-holdener commented on client_app/src/components/shelter/Schedule.tsx:
```diff
@@ -0,0 +1,104 @@
+          end: new Date(shift.end_time),
+        }))}
+        defaultView={Views.WEEK}
+        style={{ height: 500 }}
```
Please move the styling into a .css file (same comment as before)

kate-holdener reviewed (changes requested):
@Taktar were you able to run the application without Docker as client_app and server separately? There are some errors I am seeing in the client_app when I run it (details are in my review comments). Please address those. Do reach out on slack if you are running into problems running this application without Docker.

Taktar commented:
<img width="1434" alt="Screenshot 2025-02-23 at 3 50 38 PM" src="https://github.com/user-attachments/assets/3d42238e-ef79-4d20-9bbb-005e85fdc0e4" />
<img width="1435" alt="Screenshot 2025-02-23 at 3 51 15 PM" src="https://github.com/user-attachments/assets/f315d964-23f5-4e8f-87a7-4f2d5a055dcc" />
<img width="682" alt="Screenshot 2025-02-23 at 3 54 38 PM" src="https://github.com/user-attachments/assets/fb288502-f10e-446f-8041-c448db8c079b" />

kate-holdener commented:
@Taktar I get the following errors when I run the client_app. Did you commit all your changes?
```
Compiled with problems:
×
ERROR in ./src/components/shelter/Schedule.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/its_admin/projects/shelter_volunteers/client_app/src/components/shelter/Schedule.js: Unexpected token (19:63)

  17 |
  18 | const Schedule = () => {
> 19 |   const [scheduledShifts, setScheduledShifts] = useState<Shift[]>([]);
     |                                                                ^
  20 |   // The active shift type that the admin has selected for placement.
  21 |   const [activeShiftType, setActiveShiftType] = useState<string | null>(null);
  22 |
    at constructor (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:353:19)
    at FlowParserMixin.raise (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:3277:19)
    at FlowParserMixin.unexpected (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:3297:16)
    at FlowParserMixin.parseExprAtom (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10974:16)
    at FlowParserMixin.parseExprAtom (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:6821:20)
    at FlowParserMixin.parseExprSubscripts (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10590:23)
    at FlowParserMixin.parseUpdate (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOps (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10410:23)
    at FlowParserMixin.parseMaybeConditional (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5686:18)
    at FlowParserMixin.parseExpressionBase (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10302:23)
    at /Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10298:39
    at FlowParserMixin.allowInAnd (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:11936:12)
    at FlowParserMixin.parseExpression (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10298:17)
    at FlowParserMixin.parseMember (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10646:28)
    at FlowParserMixin.parseSubscript (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10634:21)
    at FlowParserMixin.parseSubscript (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5787:18)
    at FlowParserMixin.parseSubscripts (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10603:19)
    at FlowParserMixin.parseSubscripts (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5753:18)
    at FlowParserMixin.parseExprSubscripts (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10594:17)
    at FlowParserMixin.parseUpdate (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOpBaseRightExpr (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10491:34)
    at FlowParserMixin.parseExprOpRightExpr (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10486:21)
    at FlowParserMixin.parseExprOp (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10453:27)
    at FlowParserMixin.parseExprOps (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10414:17)
    at FlowParserMixin.parseMaybeConditional (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5686:18)
    at /Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10318:39
    at FlowParserMixin.allowInAnd (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:11931:16)
    at FlowParserMixin.parseMaybeAssignAllowIn (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:10318:17)
    at FlowParserMixin.parseVar (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12864:91)
    at FlowParserMixin.parseVarStatement (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12710:10)
    at FlowParserMixin.parseStatementContent (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12322:23)
    at FlowParserMixin.parseStatementLike (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12239:17)
    at FlowParserMixin.parseStatementLike (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5055:24)
    at FlowParserMixin.parseStatementListItem (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12219:17)
    at FlowParserMixin.parseBlockOrModuleBlockBody (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12796:61)
    at FlowParserMixin.parseBlockBody (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12789:10)
    at FlowParserMixin.parseBlock (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:12777:10)
    at FlowParserMixin.parseFunctionBody (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:11616:24)
    at /Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5029:63
    at FlowParserMixin.forwardNoArrowParamsConversionAt (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5205:16)
    at FlowParserMixin.parseFunctionBody (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:5029:12)
    at FlowParserMixin.parseArrowExpression (/Users/its_admin/projects/shelter_volunteers/client_app/node_modules/@babel/parser/lib/index.js:11591:10)
ERROR
[eslint] 
src/components/shelter/Schedule.js
  Line 19:63:  Parsing error: Unexpected token (19:63)
```

Taktar commented:
I thought I did, but I guess it didn’t work. I just tried again!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Kate Holdener ***@***.***>
Sent: Sunday, February 23, 2025 9:57:39 PM
To: oss-slu/shelter_volunteers ***@***.***>
Cc: Taslima Aktar ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [oss-slu/shelter_volunteers] try#167 - defining repeatable shifts (PR #188)

@ Taktar I get the following errors when I run the client_app. Did you commit all your changes? Compiled with problems: × ERROR in ./src/components/shelter/Schedule. js Module build failed (from ./node_modules/babel-loader/lib/index. js): SyntaxError: 


@Taktar<https://urldefense.com/v3/__https://github.com/Taktar__;!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wh42OL1M$> I get the following errors when I run the client_app. Did you commit all your changes?

Compiled with problems:
×
ERROR in ./src/components/shelter/Schedule.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/its_admin/projects/shelter_volunteers/client_app/src/components/shelter/Schedule.js: Unexpected token (19:63)

  17 |
  18 | const Schedule = () => {
> 19 |   const [scheduledShifts, setScheduledShifts] = useState<Shift[]>([]);
     |                                                                ^
  20 |   // The active shift type that the admin has selected for placement.
  21 |   const [activeShiftType, setActiveShiftType] = useState<string | null>(null);
  22 |
    at constructor ***@***.***/parser/lib/index.js:353:19)
    at FlowParserMixin.raise ***@***.***/parser/lib/index.js:3277:19)
    at FlowParserMixin.unexpected ***@***.***/parser/lib/index.js:3297:16)
    at FlowParserMixin.parseExprAtom ***@***.***/parser/lib/index.js:10974:16)
    at FlowParserMixin.parseExprAtom ***@***.***/parser/lib/index.js:6821:20)
    at FlowParserMixin.parseExprSubscripts ***@***.***/parser/lib/index.js:10590:23)
    at FlowParserMixin.parseUpdate ***@***.***/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary ***@***.***/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate ***@***.***/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOps ***@***.***/parser/lib/index.js:10410:23)
    at FlowParserMixin.parseMaybeConditional ***@***.***/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:5686:18)
    at FlowParserMixin.parseExpressionBase ***@***.***/parser/lib/index.js:10302:23)
    at ***@***.***/parser/lib/index.js:10298:39
    at FlowParserMixin.allowInAnd ***@***.***/parser/lib/index.js:11936:12)
    at FlowParserMixin.parseExpression ***@***.***/parser/lib/index.js:10298:17)
    at FlowParserMixin.parseMember ***@***.***/parser/lib/index.js:10646:28)
    at FlowParserMixin.parseSubscript ***@***.***/parser/lib/index.js:10634:21)
    at FlowParserMixin.parseSubscript ***@***.***/parser/lib/index.js:5787:18)
    at FlowParserMixin.parseSubscripts ***@***.***/parser/lib/index.js:10603:19)
    at FlowParserMixin.parseSubscripts ***@***.***/parser/lib/index.js:5753:18)
    at FlowParserMixin.parseExprSubscripts ***@***.***/parser/lib/index.js:10594:17)
    at FlowParserMixin.parseUpdate ***@***.***/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary ***@***.***/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate ***@***.***/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOpBaseRightExpr ***@***.***/parser/lib/index.js:10491:34)
    at FlowParserMixin.parseExprOpRightExpr ***@***.***/parser/lib/index.js:10486:21)
    at FlowParserMixin.parseExprOp ***@***.***/parser/lib/index.js:10453:27)
    at FlowParserMixin.parseExprOps ***@***.***/parser/lib/index.js:10414:17)
    at FlowParserMixin.parseMaybeConditional ***@***.***/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:5686:18)
    at ***@***.***/parser/lib/index.js:10318:39
    at FlowParserMixin.allowInAnd ***@***.***/parser/lib/index.js:11931:16)
    at FlowParserMixin.parseMaybeAssignAllowIn ***@***.***/parser/lib/index.js:10318:17)
    at FlowParserMixin.parseVar ***@***.***/parser/lib/index.js:12864:91)
    at FlowParserMixin.parseVarStatement ***@***.***/parser/lib/index.js:12710:10)
    at FlowParserMixin.parseStatementContent ***@***.***/parser/lib/index.js:12322:23)
    at FlowParserMixin.parseStatementLike ***@***.***/parser/lib/index.js:12239:17)
    at FlowParserMixin.parseStatementLike ***@***.***/parser/lib/index.js:5055:24)
    at FlowParserMixin.parseStatementListItem ***@***.***/parser/lib/index.js:12219:17)
    at FlowParserMixin.parseBlockOrModuleBlockBody ***@***.***/parser/lib/index.js:12796:61)
    at FlowParserMixin.parseBlockBody ***@***.***/parser/lib/index.js:12789:10)
    at FlowParserMixin.parseBlock ***@***.***/parser/lib/index.js:12777:10)
    at FlowParserMixin.parseFunctionBody ***@***.***/parser/lib/index.js:11616:24)
    at ***@***.***/parser/lib/index.js:5029:63
    at FlowParserMixin.forwardNoArrowParamsConversionAt ***@***.***/parser/lib/index.js:5205:16)
    at FlowParserMixin.parseFunctionBody ***@***.***/parser/lib/index.js:5029:12)
    at FlowParserMixin.parseArrowExpression ***@***.***/parser/lib/index.js:11591:10)
ERROR
[eslint]
src/components/shelter/Schedule.js
  Line 19:63:  Parsing error: Unexpected token (19:63)


—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677385425__;Iw!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wZzS2POw$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/BFOPC4CATQUXQ2ZOKMK4RGT2RKKDHAVCNFSM6AAAAABXV5M6IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZXGM4DKNBSGU__;!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wXP7FIPE$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[kate-holdener]kate-holdener left a comment (oss-slu/shelter_volunteers#188)<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677385425__;Iw!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wZzS2POw$>

@Taktar<https://urldefense.com/v3/__https://github.com/Taktar__;!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wh42OL1M$> I get the following errors when I run the client_app. Did you commit all your changes?

Compiled with problems:
×
ERROR in ./src/components/shelter/Schedule.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/its_admin/projects/shelter_volunteers/client_app/src/components/shelter/Schedule.js: Unexpected token (19:63)

  17 |
  18 | const Schedule = () => {
> 19 |   const [scheduledShifts, setScheduledShifts] = useState<Shift[]>([]);
     |                                                                ^
  20 |   // The active shift type that the admin has selected for placement.
  21 |   const [activeShiftType, setActiveShiftType] = useState<string | null>(null);
  22 |
    at constructor ***@***.***/parser/lib/index.js:353:19)
    at FlowParserMixin.raise ***@***.***/parser/lib/index.js:3277:19)
    at FlowParserMixin.unexpected ***@***.***/parser/lib/index.js:3297:16)
    at FlowParserMixin.parseExprAtom ***@***.***/parser/lib/index.js:10974:16)
    at FlowParserMixin.parseExprAtom ***@***.***/parser/lib/index.js:6821:20)
    at FlowParserMixin.parseExprSubscripts ***@***.***/parser/lib/index.js:10590:23)
    at FlowParserMixin.parseUpdate ***@***.***/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary ***@***.***/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate ***@***.***/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOps ***@***.***/parser/lib/index.js:10410:23)
    at FlowParserMixin.parseMaybeConditional ***@***.***/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:5686:18)
    at FlowParserMixin.parseExpressionBase ***@***.***/parser/lib/index.js:10302:23)
    at ***@***.***/parser/lib/index.js:10298:39
    at FlowParserMixin.allowInAnd ***@***.***/parser/lib/index.js:11936:12)
    at FlowParserMixin.parseExpression ***@***.***/parser/lib/index.js:10298:17)
    at FlowParserMixin.parseMember ***@***.***/parser/lib/index.js:10646:28)
    at FlowParserMixin.parseSubscript ***@***.***/parser/lib/index.js:10634:21)
    at FlowParserMixin.parseSubscript ***@***.***/parser/lib/index.js:5787:18)
    at FlowParserMixin.parseSubscripts ***@***.***/parser/lib/index.js:10603:19)
    at FlowParserMixin.parseSubscripts ***@***.***/parser/lib/index.js:5753:18)
    at FlowParserMixin.parseExprSubscripts ***@***.***/parser/lib/index.js:10594:17)
    at FlowParserMixin.parseUpdate ***@***.***/parser/lib/index.js:10573:21)
    at FlowParserMixin.parseMaybeUnary ***@***.***/parser/lib/index.js:10551:23)
    at FlowParserMixin.parseMaybeUnaryOrPrivate ***@***.***/parser/lib/index.js:10405:61)
    at FlowParserMixin.parseExprOpBaseRightExpr ***@***.***/parser/lib/index.js:10491:34)
    at FlowParserMixin.parseExprOpRightExpr ***@***.***/parser/lib/index.js:10486:21)
    at FlowParserMixin.parseExprOp ***@***.***/parser/lib/index.js:10453:27)
    at FlowParserMixin.parseExprOps ***@***.***/parser/lib/index.js:10414:17)
    at FlowParserMixin.parseMaybeConditional ***@***.***/parser/lib/index.js:10387:23)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:10348:21)
    at FlowParserMixin.parseMaybeAssign ***@***.***/parser/lib/index.js:5686:18)
    at ***@***.***/parser/lib/index.js:10318:39
    at FlowParserMixin.allowInAnd ***@***.***/parser/lib/index.js:11931:16)
    at FlowParserMixin.parseMaybeAssignAllowIn ***@***.***/parser/lib/index.js:10318:17)
    at FlowParserMixin.parseVar ***@***.***/parser/lib/index.js:12864:91)
    at FlowParserMixin.parseVarStatement ***@***.***/parser/lib/index.js:12710:10)
    at FlowParserMixin.parseStatementContent ***@***.***/parser/lib/index.js:12322:23)
    at FlowParserMixin.parseStatementLike ***@***.***/parser/lib/index.js:12239:17)
    at FlowParserMixin.parseStatementLike ***@***.***/parser/lib/index.js:5055:24)
    at FlowParserMixin.parseStatementListItem ***@***.***/parser/lib/index.js:12219:17)
    at FlowParserMixin.parseBlockOrModuleBlockBody ***@***.***/parser/lib/index.js:12796:61)
    at FlowParserMixin.parseBlockBody ***@***.***/parser/lib/index.js:12789:10)
    at FlowParserMixin.parseBlock ***@***.***/parser/lib/index.js:12777:10)
    at FlowParserMixin.parseFunctionBody ***@***.***/parser/lib/index.js:11616:24)
    at ***@***.***/parser/lib/index.js:5029:63
    at FlowParserMixin.forwardNoArrowParamsConversionAt ***@***.***/parser/lib/index.js:5205:16)
    at FlowParserMixin.parseFunctionBody ***@***.***/parser/lib/index.js:5029:12)
    at FlowParserMixin.parseArrowExpression ***@***.***/parser/lib/index.js:11591:10)
ERROR
[eslint]
src/components/shelter/Schedule.js
  Line 19:63:  Parsing error: Unexpected token (19:63)


—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677385425__;Iw!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wZzS2POw$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/BFOPC4CATQUXQ2ZOKMK4RGT2RKKDHAVCNFSM6AAAAABXV5M6IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZXGM4DKNBSGU__;!!K543PA!NmA_YPlblczmz-TFMrGKiUy5SLgKuBg6qzaThHCx_I-FEYgRi8lN6Lpz_kuFjJ2vp5nzmviemuBKfyD5pn5wXP7FIPE$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

kate-holdener commented:
@Taktar The error with running the application was my error. Sorry about that. This is great progress. I see that I can click on the pre-defined shifts and they will appear in the schedule under the first day. We talked about creating an "Open" button for each day, and once a user clicks on that button, the pre-defined shifts would get loaded for that day into the calendar. I took a screenshot of what we have so far, and edited it as a prototype of what we want.
Could you make this happen?

<img width="1355" alt="Screenshot 2025-02-23 at 10 28 14 PM" src="https://github.com/user-attachments/assets/ccc948d5-94f5-473b-b5a9-2395ab027233" />

The green boxes in the calendar would correspond to the pre-defined shifts. They would include the standard shift details (times, number of workers)

Taktar commented:
Good morning!

Yes, I can try to make this happen! I just wanted to have something working since I totally messed up last time. Thank you for including the screenshotting, I will use it as a reference.

Best,
Taslima

[Saint Louis University]

Taslima Aktar she/her
Saint Louis University Class of 2025
College of Arts and Sciences
Computer Science Major


Phone: 314-398-5048

***@***.***
slu.edu<https://www.slu.edu/> | slubillikens.com<https://slubillikens.com/>

________________________________
From: Kate Holdener ***@***.***>
Sent: Sunday, February 23, 2025 10:33 PM
To: oss-slu/shelter_volunteers ***@***.***>
Cc: Taslima Aktar ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [oss-slu/shelter_volunteers] try#167 - defining repeatable shifts (PR #188)

@ Taktar The error with running the application was my error. Sorry about that. This is great progress. I see that I can click on the pre-defined shifts and they will appear in the schedule under the first day. We talked about creating an "Open"


@Taktar<https://urldefense.com/v3/__https://github.com/Taktar__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgraxtV2H8$> The error with running the application was my error. Sorry about that. This is great progress. I see that I can click on the pre-defined shifts and they will appear in the schedule under the first day. We talked about creating an "Open" button for each day, and once a user clicks on that button, the pre-defined shifts would get loaded for that day into the calendar. I took a screenshot of what we have so far, and edited it as a prototype of what we want.
Could you make this happen?

Screenshot.2025-02-23.at.10.28.14.PM.png (view on web)<https://urldefense.com/v3/__https://github.com/user-attachments/assets/ccc948d5-94f5-473b-b5a9-2395ab027233__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrB2OrWwg$>

The green boxes in the calendar would correspond to the pre-defined shifts. They would include the standard shift details (times, number of workers)

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677416687__;Iw!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrzUk3NS4$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/BFOPC4HBDZ2Z6QDQTGVDJM32RKOJ5AVCNFSM6AAAAABXV5M6IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZXGQYTMNRYG4__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrG4bXDOM$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[kate-holdener]kate-holdener left a comment (oss-slu/shelter_volunteers#188)<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677416687__;Iw!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrzUk3NS4$>

@Taktar<https://urldefense.com/v3/__https://github.com/Taktar__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgraxtV2H8$> The error with running the application was my error. Sorry about that. This is great progress. I see that I can click on the pre-defined shifts and they will appear in the schedule under the first day. We talked about creating an "Open" button for each day, and once a user clicks on that button, the pre-defined shifts would get loaded for that day into the calendar. I took a screenshot of what we have so far, and edited it as a prototype of what we want.
Could you make this happen?

Screenshot.2025-02-23.at.10.28.14.PM.png (view on web)<https://urldefense.com/v3/__https://github.com/user-attachments/assets/ccc948d5-94f5-473b-b5a9-2395ab027233__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrB2OrWwg$>

The green boxes in the calendar would correspond to the pre-defined shifts. They would include the standard shift details (times, number of workers)

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/oss-slu/shelter_volunteers/pull/188*issuecomment-2677416687__;Iw!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrzUk3NS4$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/BFOPC4HBDZ2Z6QDQTGVDJM32RKOJ5AVCNFSM6AAAAABXV5M6IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZXGQYTMNRYG4__;!!K543PA!KmBn5XzSPAcPkuK2iggY2iJbPUv4CIuzVaKoHuajlWhnvs8jRDokrzaF8Up9xjXa5tOvzmx5HnF8dSkOZKgrG4bXDOM$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Taktar commented:
What Was Changed?
Schedule.jsx:

Added logic so that when an "Open Shift" button for a given day is clicked, that day’s standard shifts are loaded, and the day is recorded as "opened."
Filtered the "Open Shift" events so that once a day is opened, its button does not appear again (preventing duplicate loading).
Kept the existing functionality for single shift–clicks and calendar events unchanged.
Schedule.css:

Updated the container styles so that the calendar area is larger and scrollable, allowing the user to see more hours at once.
Why Was It Changed?
Prevent Duplicate Shifts:
To avoid multiple loads of the same standard shifts for a day if the admin clicks the "Open Shift" button repeatedly, we track which days have been opened and filter out their button events.

Improve Usability:
Increasing the calendar container’s height and enabling scrolling makes it easier for users to see more of the schedule at once.

How Was It Changed?
In Schedule.jsx:
A new state variable (openedDays) is used to record each day (using its midnight timestamp) that has been opened.
The handleOpenDate function now checks if a day has already been opened before adding shifts.
When generating the "Open Shift" events (all-day events) for the current week, we filter out days already opened so the button is not shown again.
In Schedule.css:
The .schedule-container and .calendar-container styles were updated to use a flex layout and overflow-y: auto, respectively, so the calendar can grow and be scrollable.


<img width="1413" alt="Screenshot 2025-02-24 at 12 03 44 PM" src="https://github.com/user-attachments/assets/6843cb90-ab6d-4ab4-8126-726dfe2d85d1" />
<img width="1424" alt="Screenshot 2025-02-24 at 12 03 57 PM" src="https://github.com/user-attachments/assets/de524837-18af-4526-84ad-7cf51bd25f6c" />
<img width="1428" alt="Screenshot 2025-02-24 at 12 04 22 PM" src="https://github.com/user-attachments/assets/713e42a4-29b4-474f-8ff8-6b7f7bf5194e" />
<img width="1426" alt="Screenshot 2025-02-24 at 12 04 35 PM" src="https://github.com/user-attachments/assets/f27f0462-8064-4de3-9e72-798124556cc3" />
<img width="1429" alt="Screenshot 2025-02-24 at 12 09 41 PM" src="https://github.com/user-attachments/assets/6e8c0566-fcce-44cb-9908-bd6b6175e22c" />

Taktar commented:
thats so weird. I am having a lot of trouble trying to make it work..

kate-holdener commented:
I think it's a problem in the react-big-calendar. Let's merge your work and deal with cross-midnight shifts later.
